edge-hub/src/Microsoft.Azure.Devices.Routing.Core/endpoints/StoringAsyncEndpointExecutor.cs

```diff
@@ -144,10 +151,11 @@ public async Task UpdatePriorities(IList<uint> priorities, Option<Endpoint> newE
             // Update priorities by merging the new ones with the existing.
             // We don't ever remove stale priorities, otherwise stored messages
             // pending for a removed priority will never get sent.
-            SortedDictionary<uint, EndpointExecutorFsm> snapshot = this.prioritiesToFsms;
+            ImmutableDictionary<uint, EndpointExecutorFsm> snapshot = this.prioritiesToFsms;
+            Dictionary<uint, EndpointExecutorFsm> updatedSnapshot = new Dictionary<uint, EndpointExecutorFsm>(snapshot);

```

Comment:
>Dictionary [](start = 12, length = 10)

why not use a SortedDictionary as the updated snapshot?  You can still convert it to an ImmutableDictionary at the end when writing back to the AtomicReference.  This way the sorting is happening on insertion, and we wouldn't have to re-sort on every iteration of the message pump loop.

Reply:
It is because ImmutableDictionary is not guaranteed to be sorted, and we should not depend on internal implementation how immutableDictionary persist the data.  Therefore uses Dictionary here and do sorting during consumption.